crates/wasi-provider/src/lib.rs

```diff
@@ -18,22 +18,35 @@ use handle::PodHandle;
 use wasi_runtime::WasiRuntime;
 
 const TARGET_WASM32_WASI: &str = "wasm32-wasi";
+const LOG_DIR_NAME: &str = "wasi-logs";
 
 /// WasiProvider provides a Kubelet runtime implementation that executes WASM
 /// binaries conforming to the WASI spec
 #[derive(Clone)]
 pub struct WasiProvider {
     handles: Arc<RwLock<HashMap<String, PodHandle<File>>>>,
     store: FileModuleStore,
+    log_path: PathBuf,
 }
 
 impl WasiProvider {
-    pub fn new() -> Self {
-        let store = FileModuleStore::new(&dirs::home_dir().expect("Cannot get home directory"));
-        Self {
+    /// Returns a new WASI provider configured to use the proper data directory
+    /// (including creating it if necessary)
+    pub fn new<P: AsRef<Path>>(data_dir: P) -> anyhow::Result<Self> {
+        // Make sure we have a log dir and containers dir created
+        let data_path = data_dir.as_ref().to_path_buf();
```

Comment:
We should probably just take a `PathBuf` if we're just going to turn it into one anyway

Reply:
Ha, I predicted this comment would come 😄. So my reasoning here is that it makes this clean for testing or hard coded paths (when someone just wants to use a `&str`) and still allows us to use the `PathBuf` from the config. My goal was to make this as easy as possible for users (e.g. just passing in anything that implements `AsRef<Path>`) who use this crate.

Reply:
That's fine.  If we want ultimate flexibility this is ok.